string[] nodes_idx
string[] links_idx
